gemrb/GUIScripts/InventoryCommon.py

```diff
@@ -829,24 +825,20 @@ def GetColor():
 	if GameCheck.IsIWD2 () or GameCheck.IsGemRBDemo ():
 		Button = ColorPicker.GetControl (35)
 		Button.OnPress (CancelColor)
-		Button.SetText (103)
 		if GameCheck.IsIWD2 ():
 			Button.SetText (13727)
+		else:
+			Button.SetText (103)
 
 	for i in range (34):
 		Button = ColorPicker.GetControl (i)
 		MyColor = ColorTable.GetValue (ColorIndex, i)
 		if MyColor == "*":
 			Button.SetState (IE_GUI_BUTTON_LOCKED)
 			continue
-		if PickedColor == MyColor:
-			GemRB.SetVar ("Selected",i)
-			Button.SetState (IE_GUI_BUTTON_LOCKED)
-			Button.MakeEscape()
-		else:
-			Button.SetBAM ("COLGRAD", 2, 0, MyColor)
-			Button.SetFlags (IE_GUI_BUTTON_PICTURE|IE_GUI_BUTTON_RADIOBUTTON, OP_OR)
-			Button.SetState (IE_GUI_BUTTON_ENABLED)
+		Button.SetBAM ("COLGRAD", 2, 0, MyColor)
+		Button.SetFlags (IE_GUI_BUTTON_PICTURE|IE_GUI_BUTTON_RADIOBUTTON, OP_OR)
+		Button.SetState (IE_GUI_BUTTON_ENABLED)
```

Comment:
Are these changes enough for the selected color button to actually appear selected (green outline)? If not it needs to be in a different state.

Reply:
Oh, thanks. We have some inconsistency here. BG1 doesn't use the green outline and doesn't indicate the currently selected color:
![bg1_wine](https://github.com/gemrb/gemrb/assets/155195419/54175e2e-d556-4076-b28c-d1823dd4eb57)

We have a BG2 GameCheck in GUICG13 to set the buttons to radiobuttons. In GUIREC it doesn't set the buttons to radiobuttons. In InventoryCommon we set the buttons to radiobuttons unconditionally.

This brings me to the following conclusion:
- We need BG2 GameChecks in all three places and turn the buttons into radiobuttons for BG2. For BG1 these should not be radiobuttons.
- We need to apply the radiobutton fix we already did for the sound frequency: set the visible state of the currently selected button to IE_GUI_BUTTON_SELECTED and adjust the dictionary with an explicit `GemRB.SetVar("Selected", pickedColorIdx)`. Therefore we need PickedColor as a global variable again.

Does this sound correct to you?

Reply:
Yes and no, let's just do the same in bg1 if possible. If the button bams don't have a selected frame, too bad, we'll need to work around that, but if they do, having a visual indicator of what is the current choice is good.

Reply:
As far as I can see there is no BAM for a selected frame. I tried to unify the button handling and used radiobuttons for BG1 too, but that doesn't look good:

![bg1_gemrb_000](https://github.com/gemrb/gemrb/assets/155195419/39904cb2-2757-4d66-9849-30c26646b83e)

You see that we can have some kind of broken image as an indicator, but I wouldn't do it. There's also some different handling of the button state between BG1 and 2. 2 disables unused buttons and 1 locks unused buttons. Without testing BG2 I don't know how that behaves. All in all I think we don't get around doing things differently for BG1.

Now I also realize that the position of that color picker window is wrong and there are three colors missing in the last row.

Reply:
Yeah, offsetting the position will help with knowing what is already selected. Still, bg2 would need to be tested. But the whole thing is tangled in other ongoing work, so meh, not sure what's the best way to continue.

Reply:
I wouldn't think too much about adding a color selection indicator for BG1, it wasn't there in the originals and people hadn't problems with it ;)

Yeah, let me look into the GUIScript-refactor branch. Maybe putting the bugfixes into this branch is the right way to go forward and lay this PR to rest. Let's see.